be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/MessageHandler.scala

```diff
@@ -92,7 +91,7 @@ trait MessageHandler extends AskPatternConstants {
     * @return
     *   the database answer wrapped in a [[scala.concurrent.Future]]
     */
-  def dbBroadcast(rpcMessage: JsonRpcRequest, channel: Channel, broadcastData: Base64Data, broadcastChannel: Channel): Future[GraphMessage] = {
+  def dbBroadcast(rpcMessage: JsonRpcRequest, channel: Channel, broadcastData: String, broadcastChannel: Channel): Future[GraphMessage] = {
```

Comment:
Since the `String` type here allows to pass garbage in this function much more easily (even if only by mistake), and that every caller pretty much does `.toJson.toString` , would it make sense to take a JSON as parameter instead ?

Reply:
I guess we can also do that. Indeed, we always call `.toString` after the JSON encoding of the data. 
It is a way to generalize the `dbBroadcast` function. 
We can also still stick with the initial implementation where the `Base64.encode` is outside of the function.
We have two options. I do not really see a more favourable one, except the generalization argument.

Reply:
I see your point about passing something wrong to the function. The `Base64.encode` did not ensure anything more reliable as well (with the previous implementation of the function, where the function took a Base64 Hash instead of a String or JSON value), we could have also passed anything wrong that was simply encoded I think.

Reply:
I agree with you, we didn't ensure anything more coherent. However, the intention of what was supposed to go in there was a little bit more obvious (and it is again with the `JsValue`). Basically, type safety wasn't so much the key point as much as giving a good `tip` to the developer as to what was expected 😁